app/models/reporting/caseworker_report.rb

```diff
@@ -1,89 +1,30 @@
+      @dataset = dataset
     end
 
-    def rows
-      dataset.values.sort_by { |h| h[:user_name].upcase }
-    end
+    def rows(sorting: CaseworkerReportSorting.default)
+      sorted_rows = dataset.values.sort_by do |r|
+        v = r.public_send(sorting.sort_by)
+        v.respond_to?(:upcase) ? v.upcase : v
+      end
 
-    private
+      return sorted_rows unless sorting.sort_direction == 'descending'
 
-    def dataset
-      @dataset ||= load_from_events
+      sorted_rows.reverse
     end
 
-    def load_from_events # rubocop:disable  Metrics
-      @scope.init(-> { {} })
-            .when(
-              Assigning::AssignedToUser,
-              lambda { |rows, event|
-                user_id = event.data.fetch(:to_whom_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_assigned_to_user] += 1
-              }
-            )
-            .when(
-              Assigning::ReassignedToUser,
-              lambda { |rows, event|
-                user_id = event.data.fetch(:to_whom_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_assigned_to_user] += 1
-
-                user_id = event.data.fetch(:from_whom_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_unassigned_from_user] += 1
-              }
-            )
-            .when(
-              Assigning::UnassignedFromUser,
-              lambda { |rows, event|
-                user_id = event.data.fetch(:from_whom_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_unassigned_from_user] += 1
-              }
-            )
-            .when(
-              Reviewing::SentBack,
-              lambda { |rows, event|
-                user_id = event.data.fetch(:user_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_closed_by_user] += 1
-              }
-            )
-            .when(
-              Reviewing::Completed,
-              lambda { |rows, event|
-                user_id = event.data.fetch(:user_id)
-                rows[user_id] ||= initial_user_row(user_id)
-                rows[user_id][:total_closed_by_user] += 1
-              }
-            )
-            .run(Rails.application.config.event_store)
-    end
+    private
 
-    # returns a hash of the initial row columns with values set to zero
-    def initial_user_row(user_id)
-      {
-        user_name: User.name_for(user_id),
-        total_assigned_to_user: 0,
-        total_unassigned_from_user: 0,
-        total_closed_by_user: 0
-      }
-    end
+    attr_reader :dataset
 
     class << self
       def for_temporal_period(date:, interval:)
-        stream_name = date.strftime(
-          CaseworkerReports::Configuration::STREAM_NAME_FORMATS.fetch(interval)
-        )
+        stream_name = CaseworkerReports.stream_name(date:, interval:)
```

Comment:
I wonder if stream name is confusing now we've added work streams? not blocking just a thought

Reply:
Think this could become an issue, but I suggest we have a relook at it this if it becomes confusing a bit later on :)